### PR TITLE
fix: menu horizontal dark style & site

### LINKED ISSUE #25845: Menu with dark theme: selected item's background should have padding around content
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-byjl1?file=/index.js)

### Steps to reproduce
Use `Menu` component with `theme="dark"` and select one of the menu items. 

### What is expected?
The background color of the selected navigation item should extend to the left and right of the item's content. 

Before 4.5.0:
Screen Shot 2020-07-27 at 12 47 04 PM

### What is actually happening?
The background color ends abruptly to the left and right of the selected item's content. 

After 4.5.0:
Screen Shot 2020-07-27 at 12 47 23 PM

| Environment | Info |
|---|---|
| antd | 4.5.0 |
| React | 16.12.0 |
| System | any |
| Browser | any |

---
This is due to the following change: https://github.com/ant-design/ant-design/pull/25622

With the `light` theme, the padding is not needed, because it makes sense for the underline to end where the text ends. But with the `dark` theme, it looks very unnatural without the padding.

### PR DESCRIPTION
**Pull request by 07akioni**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix https://github.com/ant-design/ant-design/issues/25845
ref https://github.com/ant-design/ant-design/commit/f0509f0bf96e47b31ee4616f7db5179bdc737f83#comments

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix Menu in `horizontal` `dark` status Menu.Item background style glitch.     |
| 🇨🇳 Chinese |   修复 Menu 在 `horizontal` `dark` 状态下 Menu.Item 背景样式出错。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/menu/demo/style-debug.md](https://github.com/ant-design/ant-design/blob/fix-menu-dark-style/components/menu/demo/style-debug.md)

### CODE REVIEW

**Comment by afc163** [comment]
紧凑版也看看。

**Comment by xrkffgg** [comment]
RTL 别忘了哟

File: site/theme/template/Layout/Header/Navigation.less [hunk 1]
```diff
@@ -13,9 +13,11 @@
 
     & > .ant-menu-item,
     & > .ant-menu-submenu {
+      min-width: 72px;
       height: @header-height;
-      margin-right: 12px;
-      margin-left: 12px;
+      margin: 0;
+      padding-right: 12px;
+      padding-left: 12px;
```
Comment (afc163) [inline site/theme/template/Layout/Header/Navigation.less:20]: 这里我 master 里修复了，不用动。

Comment (07akioni) [reply site/theme/template/Layout/Header/Navigation.less:20]: master  那个修复的和原来不一样，你对比一下老的

Comment (afc163) [reply site/theme/template/Layout/Header/Navigation.less:20]: 不和老的对了，我相应改了一下 UI。

**Comment by 07akioni** [comment]
紧凑主题和 Dark 主题的 dark menu 预览还是存在问题。不过 dist/antd.dark.js 是没啥问题，应该是 work 的。

给构建工具提了个 issue

ref https://github.com/chenshuai2144/antd-pro-merge-less/issues/21

**Review by afc163 — APPROVED** [review]

**Pull request merged**